charger/easee.go

```diff
@@ -379,22 +384,108 @@ func (c *Easee) Enable(enable bool) error {
 		}
 
 		uri := fmt.Sprintf("%s/chargers/%s/settings", easee.API, c.charger)
-		resp, err := c.Post(uri, request.JSONContent, request.MarshalJSON(data))
-		if err != nil {
+		if err := c.postJSONAndWait(uri, false, request.MarshalJSON(data)); err != nil {
 			return err
 		}
-		resp.Body.Close()
 	}
 
 	// resume/stop charger
 	action := easee.ChargePause
 	if enable {
 		action = easee.ChargeResume
 	}
+
 	uri := fmt.Sprintf("%s/chargers/%s/commands/%s", easee.API, c.charger, action)
-	_, err := c.Post(uri, request.JSONContent, nil)
+	if err := c.postJSONAndWait(uri, true, nil); err != nil {
+		return err
+	}
 
-	return err
+	if enable {
+		// reset currents after enable, as easee automatically resets to maxA
+		return c.MaxCurrent(int64(c.current))
+	}
+
+	return nil
+}
+
+// posts JSON to the Easee API endpoint and waits for the async response
+func (c *Easee) postJSONAndWait(uri string, isCommand bool, data io.ReadSeeker) error {
+	for retriesLeft := 2; retriesLeft >= 0; retriesLeft-- {
+		resp, err := c.Post(uri, request.JSONContent, data)
+		if err != nil {
+			return err
+		}
+		defer resp.Body.Close()
+
+		if resp.StatusCode == 200 { //sync call
+			return nil
+		}
+
+		if resp.StatusCode == 202 { //async call, wait for response
+			var cmd easee.RestCommandResponse
+
+			if isCommand { //command endpoint
+				if err := decodeJSON(resp, &cmd); err != nil {
+					return err
+				}
+			} else { //settings endpoint
+				var cmdArr []easee.RestCommandResponse
+				if err := decodeJSON(resp, &cmdArr); err != nil {
+					return err
+				}
+
+				if len(cmdArr) != 0 {
+					cmd = cmdArr[0]
+				}
+			}
+
+			if cmd.Ticks == 0 { //Easee API ignored this call, retry
+				c.log.DEBUG.Printf("api call ignored, %d retries left", retriesLeft)
+				if _, err := data.Seek(0, io.SeekStart); err != nil {
+					return err
+				}
+				time.Sleep(time.Second)
+				continue
+			}
+			return c.waitForTickResponse(cmd.Ticks)
+		}
+
+		// all other response codes lead to an error
+		return fmt.Errorf("unexpected HTTP result code %d, response: %v", resp.StatusCode, resp)
+	}
+
+	// retries exhausted
+	return errors.New("retries exhausted, API call failed")
```

Comment:
```suggestion
	return api.ErrTimeout
```

Reply:
Hatte in api/error.go geschaut aber nichts passendes gefunden. ErrTimeout finde ich hier eher irreführend. Ist ja kein Timeout sondern "Ich geb auf, die API spielt nicht mit". Sowas wie BadResponse wäre passend, müßten wir dann einführen.

Reply:
Timeout=keine Antwort innerhalb der Wartezeit